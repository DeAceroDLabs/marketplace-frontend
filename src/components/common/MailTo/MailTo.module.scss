@use "styles/colors";
@use "styles/typography";

.mail{
    color: colors.$accent;
    background-color: transparent;
}

.mail:visited {
    color: colors.$accent;
    background-color: transparent;
}

.mail:active {
    color: colors.$accent;
}