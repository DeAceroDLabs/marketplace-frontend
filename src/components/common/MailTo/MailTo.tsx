import styles from "./MailTo.module.scss";

interface MailToProps {
  children?: React.ReactNode;
  email: string;
}

const MailTo: React.FC<MailToProps> = ({ children, email }) => {
  return (
    <a className={styles.mail} href={`mailto:${email}`}>
      {children}
    </a>
  );
};

export default MailTo;
