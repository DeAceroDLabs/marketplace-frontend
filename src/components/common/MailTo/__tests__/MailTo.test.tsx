import renderer from "react-test-renderer";
import MailTo from "../MailTo";

const MailToMockProps = {
  title: "Mock Title",
  imgSrc: "./mock/image/route",
};
it("renders MailTo with no issue", () => {
  const component = renderer.create(<MailTo email="contact@example.com" subject= "Hello" body="hello"/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
