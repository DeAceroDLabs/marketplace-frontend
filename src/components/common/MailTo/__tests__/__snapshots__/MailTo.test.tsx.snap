// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MailTo with no issue 1`] = `
<a
  className="mail"
  href="mailto:contact@example.com"
/>
`;
